compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -393,13 +393,23 @@ impl<'hir> LoweringContext<'_, 'hir> {
     // If `cond` kind is `let`, returns `let`. Otherwise, wraps and returns `cond`
     // in a temporary block.
     fn manage_let_cond(&mut self, cond: &'hir hir::Expr<'hir>) -> &'hir hir::Expr<'hir> {
+        let mut wrap = || {
+            let span_block =
+                self.mark_span_with_reason(DesugaringKind::CondTemporary, cond.span, None);
+            self.expr_drop_temps(span_block, cond, AttrVec::new())
+        };
         match cond.kind {
-            hir::ExprKind::Let(..) => cond,
-            _ => {
-                let span_block =
-                    self.mark_span_with_reason(DesugaringKind::CondTemporary, cond.span, None);
-                self.expr_drop_temps(span_block, cond, AttrVec::new())
+            hir::ExprKind::Binary(_, lhs, rhs) => {
```

Comment:
This check should be recursive: it should check if either side is an let chain rather than simply a let expression. I think that something like `if let x= .. && let y = .. && cond` won't work currently.

Reply:
I will add recursion but `if let x= .. && let y = .. && cond` is working with the current logic -> https://github.com/rust-lang/rust/pull/88642/files#diff-c613ee6b091cf39afd840d0f4350a6a0c819b193ecde798c1ab8c5312810c644

Can you provide another snippet?

Reply:
The body needs to use `y`

Reply:
Oh, I see.

Just fixed the issue, thanks